crabs/detection_tracking/inference_model.py

```diff
@@ -96,27 +105,35 @@ def load_video(self) -> None:
         if not self.video.isOpened():
             raise Exception("Error opening video file")
 
-        # read input video parameters
-        frame_width = int(self.video.get(cv2.CAP_PROP_FRAME_WIDTH))
-        frame_height = int(self.video.get(cv2.CAP_PROP_FRAME_HEIGHT))
-        cap_fps = self.video.get(cv2.CAP_PROP_FPS)
-
         # prepare output video writer if required
         if self.args.save_video:
-            output_file = f"{self.video_file_root}output_video.mp4"
-            output_codec = cv2.VideoWriter_fourcc(*"mp4v")
+            # read input video parameters
+            frame_width = int(self.video.get(cv2.CAP_PROP_FRAME_WIDTH))
+            frame_height = int(self.video.get(cv2.CAP_PROP_FRAME_HEIGHT))
+            cap_fps = self.video.get(cv2.CAP_PROP_FPS)
+            output_file = f"{self.video_file_root}_output_video.mp4"
+            output_codec = cv2.VideoWriter_fourcc(*"H264")
             self.out = cv2.VideoWriter(
                 output_file, output_codec, cap_fps, (frame_width, frame_height)
             )
 
-    def prep_csv_writer(self):
+    def prep_csv_writer(self) -> Tuple[Any, TextIO]:
         """
         Prepare csv writer to output tracking results
         """
+
+        self.tracking_output_dir = (
+            Path("crabs_tracks_label") / self.video_file_root
+        )
+        crabs_tracks_label_dir = Path("crabs_tracks_label")
+        crabs_tracks_label_dir.mkdir(parents=True, exist_ok=True)
+
+        # Create the subdirectory for the specific video file root
         self.tracking_output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
```suggestion
        crabs_tracks_label_dir = Path("crabs_tracks_label")
        self.tracking_output_dir = (
            crabs_tracks_label_dir / self.video_file_root
        )
        # Create the subdirectory for the specific video file root
        self.tracking_output_dir.mkdir(parents=True, exist_ok=True)
```

I think the order of the variables got mixed up.

I think we can do with only one `mkdir`, since it will create all of the parent directories in the path that need to be created.